### DISCUSSION TITLE
gist restrictions

### DISCUSSION THREAD

**Discussion by UwU-VPN-db** [community/community#147837] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I'm going to make my own VPN and want to store it in db on github servers as a gist.
Dear GitHub - will I have any restrictions on the number of gist or file size? (I don’t need much, about 2kb per file, and no more than 50 thousand files)

**Comment by Aplex2723** [comment, marked as answer]
GitHub allows storing data in gists with certain restrictions. Individual files are limited to 100 MB (25 MB via the web interface), and repositories are recommended to stay under 5 GB for optimal performance. Storing 50,000 files, each around 2 KB (totaling about 100 MB), is well within these limits. While there’s no strict restriction on the number of gists or files, managing such a large number might impact performance. To optimize, consider organizing data efficiently and monitoring repository size. For larger files, Git Large File Storage (LFS) offers additional options. Your use case should work fine as planned, but maintaining efficiency is key.

https://github.com/orgs/community/discussions/120943
